kasa/deviceconfig.py

```diff
@@ -131,15 +142,22 @@ class DeviceConfig:
     http_client: Optional[httpx.AsyncClient] = field(default=None, compare=False)
 
     def __post_init__(self):
-        if self.credentials is None:
-            self.credentials = Credentials(username="", password="")
         if self.connection_type is None:
             self.connection_type = ConnectionType(
                 DeviceFamilyType.IotSmartPlugSwitch, EncryptType.Xor
             )
 
-    def to_dict(self) -> Dict[str, Dict[str, str]]:
+    def to_dict(
+        self,
+        *,
+        credentials_hash: Optional[str] = None,
+        exclude_credentials: bool = False,
```

Comment:
Maybe it'd be better to always export only the credentials hash, and get rid off these arguments?

Reply:
It's needed to provide the hash as a parameter because the hash is in the transport itself and not the device config.  I think the parameters are good to have because they give the options of credentials, no credentials and hashes which covers all use cases.

Reply:
Hmmm, I think the question is should the transport details then also be stored in the exported config? It's again about trying to keep the API as simple as straightfoward as possible for downstreams, now it's sort of unclear why some parts are stored inside the config and some are passed in when exporting etc...